src/util.rs

```diff
@@ -194,15 +194,19 @@ pub(crate) async fn peer_connected(socket: tokio::net::TcpStream, backend: Arc<d
 }
 
 /// Opens port described by endpoint and starts a coroutine to accept new
-/// connections on it Returns stop_handle channel that can be used to stop
-/// accepting new connections
+/// connections on it.
+///
+/// Returns stop_handle channel that can be used to stop accepting new
+/// connections. Also mutates `endpoint` to reflect the resolved endpoint
+/// address.
 pub(crate) async fn start_accepting_connections(
-    endpoint: &Endpoint,
+    endpoint: &mut Endpoint,
```

Comment:
I was a bit conflicted on how to communicate the resolved endpoint back to the caller. I could take an owned `endpoint` but then it would waste resources if this function errored (i.e. the caller might have to defensively clone the endpoint in case of failure if they wanted to try again). I could alternatively not mutate the endpoint at all, but rather return just the port number that was resolved.